mdakit_membcurv/tests/test_mdakit_membcurv.py

```diff
@@ -3,10 +3,274 @@
+                                                -1.36958801e-02, 2.75305642e-03, -3.89289835e-03,
+                                                -2.05699114e-02],
+                                               [-2.13816311e-02, -3.60467810e-03, 6.62439558e-02,
+                                                -9.62066579e-03, -2.83642243e-02, -6.27955986e-02,
+                                                -5.31456175e-02, -8.46882808e-03, 1.37664910e-02,
+                                                1.68963909e-02],
+                                               [-2.57613213e-03, -5.75152336e-03, -3.04009492e-03,
+                                                4.38720368e-02, 8.95669062e-02, -2.11138887e-03,
+                                                1.21192352e-01, 2.35164083e-02, -1.32594642e-02,
+                                                -8.60185561e-03],
+                                               [-6.55374523e-03, -7.43313172e-03, -4.29013834e-02,
+                                                -1.93519324e-03, 2.36387158e-02, 2.59636175e-02,
+                                                4.16371248e-03, -5.25782227e-03, -2.71611977e-03,
+                                                3.10811328e-05],
+                                               [-5.32693655e-03, -2.85760622e-03, -5.49568585e-04,
+                                                2.04948874e-02, -1.76805356e-02, -1.36782305e-02,
+                                                -3.16246302e-02, -5.62996471e-03, -1.44329381e-04,
+                                                -3.73865110e-02],
+                                               [6.22342127e-03, 1.94289676e-02, 2.82196188e-05,
+                                                2.69411991e-03, -1.30866347e-02, -2.52293095e-02,
+                                                -7.22431515e-03, -6.73201764e-03, -2.51286123e-01,
+                                                -3.22793339e-01],
+                                               [-2.24613530e-02, 7.82002286e-04, -3.98911117e-04,
+                                                5.29635968e-04, -4.19699155e-03, -1.30479296e-02,
+                                                -2.38928651e-04, 4.24992903e-02, 1.70628313e-02,
+                                                -4.85138439e-02],
+                                               [-1.94220106e-02, -2.13212988e-04, 1.05435416e-03,
+                                                1.30710748e-03, -3.49076390e-03, -1.28226749e-02,
+                                                -1.41168166e-02, -1.08870941e-02, -1.21323346e+00,
+                                                -1.55778010e-01]])}
+}
+
+
+@pytest.fixture()
+def universe():
+    u = mda.Universe(GRO_, XTC_)
+    return u
+
+
+@pytest.fixture()
+def ref_beads():
+    u = mda.Universe(GRO_PO4)
+    ref_beads = u.atoms[0:455]
+    return ref_beads
+
+
+@pytest.fixture()
+def Z_cloud(ref_beads):
```

Comment:
Is this code reproducing some functionality in the main code? You are doing a fair amount of calculation to set up a reference value. Can you add some commentary here to make clear what happens? 

This is even more important in the light of https://github.com/MDAnalysis/membrane-curvature/pull/13/files#r650475495 if there are concerns how PBC are treated here. If you are generated an incorrect reference and then test your code against it, you will ensure that your code is incorrect. I understand that at this stage you want to test that your code produces the values that it has been producing in the past. If during writing the tests you find potential bugs in your code then you have two options (in my opinion)

1. write correct test and [mark them XFAIL](https://docs.pytest.org/en/6.2.x/skipping.html#xfail-mark-test-functions-as-expected-to-fail) (i.e., you know your code will currently not produce the correct result) and fix later
2. write correct tests and fix your code now

If things are not clear cut then at least add a comment in the tests and in your main code noting that this is something that needs to be looked into.

Reply:
It partially does, yes. It is a short calculation to map the `(x,y)` to `[i,j]` of the beads in the grofile that contains `PO4` beads only. The subsequent calculation of membrane curvature uses the grid I am returning here, which is a `np.array`, indexed according to number of cells in the grid, and which contains the average `z` values of the beads that populate such unit cell. I will add comment on the steps. It's the same way it is calculated in `mods.py`, [line 291-303](https://github.com/MDAnalysis/membrane-curvature/blob/main/mdakit_membcurv/lib/mods.py).
My original code doesn't consider PBC conditions of any kind, and I'm aware that's a potential risk. The calculation of membrane curvature in this approach has an exception for the atoms of coordinates that when mapped to a grid, they fall in an `[i,j]` pair bigger than the initial size of the box.

Reply:
I agree with @orbeckst that I am worried this is checking for incorrect functionality. Instead of having this function and loading ref_beads from a file, could you simply define Z_cloud as the z_ref array?

The simplest way to do this would be to run the code yourself and copy the coordinates into here, then we can work up to making sure that the generation of z_ref is correct.

Reply:
You could then also write a test for the lines of code that generate `z_ref` :)